desc/geometry/core.py

```diff
@@ -143,19 +169,19 @@ def compute(
         return data
 
     def translate(self, displacement=[0, 0, 0]):
-        """Translate the curve by a rigid displacement in x, y, z."""
-        self.shift += jnp.asarray(displacement)
+        """Translate the curve by a rigid displacement in X, Y, Z."""
+        self.shift = self.shift + jnp.asarray(displacement)
 
     def rotate(self, axis=[0, 0, 1], angle=0):
```

Comment:
This syntax means that the angle kwarg has to be passed in, you can't do `Curve.rotate(axis)` and have the angle be the magnitude of the axis. That was the way the code was originally so I'm leaving it that way for now, but we might want to rethink this.

Reply:
I think being explicit about the angle is my preferred API for this